src/main/java/fileprocessor/FindFiles.java

```diff
@@ -0,0 +1,55 @@
+package fileprocessor;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.java.Log;
+import lombok.val;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Log
+@RequiredArgsConstructor
+public class FindFiles implements Callable<List<File>> {
+
+    private final String directory;
+
+    @Override
+    public List<File> call() throws Exception {
+        File folder = new File(directory);
+
+        log.fine(STR."Folder: \{folder.getAbsolutePath()}");
+
+        val files = recursivelyGetFilesToScrape(folder).collect(Collectors.toList());
```

Comment:
I couldn't figure out how to modify recursivelyGetFilesToScrape to return only files, but adding a filter to this worked enough to get me past the error.
```
        val files = recursivelyGetFilesToScrape(folder).filter(File::isFile).collect(Collectors.toList());
```